packages/constellation/src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -1,26 +1,33 @@
 /* eslint-disable import/no-unresolved */
 import '../styles.css'
 
-import React, { CSSProperties, useMemo } from 'react'
+import React, { useEffect } from 'react'
 
+import { DarkModeContext } from './themeContext'
 import { ThemeProviderComponent } from './ThemeProvider.types'
 
 const ThemeProvider: ThemeProviderComponent = ({ children, theme, colorOverride = {} }) => {
-  const colorOverrideStyles = useMemo((): CSSProperties => {
-    return Object.keys(colorOverride).reduce(
-      (newColorOverride, color) => ({
-        ...newColorOverride,
-        [`--color-${color}`]: colorOverride[color as keyof typeof colorOverride],
-      }),
-      {},
+  useEffect(() => {
+    if (theme === 'dark') {
+      document.body.classList.add('theme-dark')
+      document.body.classList.remove('theme-light')
+    } else {
+      document.body.classList.add('theme-light')
+      document.body.classList.remove('theme-dark')
+    }
+
+    const colorOverrideStyles = Object.keys(colorOverride).reduce(
+      (newColorOverride, color) =>
+        `${newColorOverride} --color-${color}: ${
+          colorOverride[color as keyof typeof colorOverride]
+        };`,
+      '',
     )
-  }, [colorOverride])
 
-  return (
-    <div className={theme === 'dark' ? 'theme-dark' : 'theme-light'} style={colorOverrideStyles}>
-      {children}
-    </div>
-  )
+    document.body.style.cssText = colorOverrideStyles
+  }, [colorOverride, theme])
+
+  return <DarkModeContext.Provider value={theme === 'dark'}>{children}</DarkModeContext.Provider>
```

Comment:
I guess I'm confused as to why this is dark mode specific. I'm having trouble understanding how this works with themes and distinguishing themes from modes. To me it looks like we have light and dark themes in the input.css, but then we also have corresponding light and dark 'modes'. Is the intention to map every theme to a mode?

If I wanted to create a new 'malibu' theme, what would be the steps to fit it into this model?

Reply:
The naming here I think is wrong. What are currently called the light and dark 'themes' should really be 'modes' and the 'colorOverrides' should be the custom theme. The context was actually just added to work around a storybook-specific bug in the colour pallet and after taking another look at it I'm not sure its really needed. Ideally as the themes and modes are only changing the colours the components shouldn't need to know anything about what theme or mode is applied as long as they are using the variables. 

To create a Malibu theme as the Theme Provider is set up now:

1. create a JSON file or an object that would include any colour you want to change, could be a while new pallet or it could just be a few accent colours
3. When you want to enable that theme you set the colorOverrides prop on the Theme Provider to your object of theme colours

Using what are currently called colorOverrides for the theme allows you to retain the ability to have a light and dark mode version of the theme. It also allows you to create list invasive themes that only change a few colours as it will just use the standard colours for anything that isnt changed.